test/middleware/srgssr.spec.js

```diff
@@ -41,12 +43,253 @@ describe('SrgSsr', () => {
+      const spyOnRemoveTrack = jest.spyOn(player.textTracks(), 'removeTrack');
+
+      player.textTracks().getTrackById.mockReturnValueOnce({});
+      SrgSsr.addChapters(player);
+
+      expect(spyOnRemoveTrack).toHaveBeenCalled();
+    });
+
+    it('should not add the chapter if the only available chapter is the main chapter', async () => {
+      const chapterUrn = 'urn:full:length';
+      const result = [];
+
+      Pillarbox.TextTrack
+        .prototype
+        .addCue
+        .mockImplementation((cue) => result.push(cue));
+
+      SrgSsr.addChapters(player, chapterUrn, [{
+        fullLengthMarkIn: 0,
+        fullLengthMarkOut: 10000
+      }]);
+
+      expect(result).toHaveLength(0);
+    });
+
+    it('should add all chapters that are not the main chapter', async () => {
+      const chapterUrn = 'urn:full:length';
+      const result = [];
+
+      Pillarbox.TextTrack
+        .prototype
+        .addCue
+        .mockImplementation((cue) => result.push(cue));
+
+      SrgSsr.addChapters(player, chapterUrn, [{
+        fullLengthMarkIn: 0,
+        fullLengthMarkOut: 10000
+      }, {
+        fullLengthUrn: chapterUrn,
+        fullLengthMarkIn: 2500,
+        fullLengthMarkOut: 5000
+      }, {
+        fullLengthUrn: chapterUrn,
+        fullLengthMarkIn: 6000,
+        fullLengthMarkOut: 9500
+      }]);
+
+      expect(result).toHaveLength(2);
+    });
+  });
+
+  /**
+   *****************************************************************************
+   * addIntervals **************************************************************
+   *****************************************************************************
+   */
+  describe('addIntervals', () => {
+    it('should not create an interval track if the intervals parameter is not an array or if the array is empty', async () => {
+      const spyOnPillarboxTextTrack = jest.spyOn(Pillarbox, 'TextTrack');
+
+      SrgSsr.addIntervals(player, true);
+      SrgSsr.addIntervals(player, null);
+      SrgSsr.addIntervals(player, '');
+      SrgSsr.addIntervals(player, undefined);
+
+      expect(spyOnPillarboxTextTrack).not.toHaveBeenCalled();
+    });
+
+    it('should remove intervals track if any', async () => {
+      const spyOnRemoveTrack = jest.spyOn(player.textTracks(), 'removeTrack');
+
+      player.textTracks().getTrackById.mockReturnValueOnce({});
+      SrgSsr.addIntervals(player);
+
+      expect(spyOnRemoveTrack).toHaveBeenCalled();
+    });
+
+    it('should add intervals to the player', async () => {
+      const result = [];
+
+      Pillarbox.TextTrack
+        .prototype
+        .addCue
+        .mockImplementation((cue) => result.push(cue));
+
+      SrgSsr.addIntervals(player, [{
+        markIn: 1_000,
+        markOut: 2_000,
+        type: 'OPENING_CREDITS'
+      }, {
+        markIn: 10_000,
+        markOut: 15_000,
+        type: 'CLOSING_CREDITS'
+      }]);
+
+      expect(result).toHaveLength(2);
```

Comment:
It could be interesting to check if the data is correctly stored in the `text` field, e.g.:

```suggestion
      expect(result).toHaveLength(2);
      expect(JSON.parse(result[0].text).type).toBe('OPENING_CREDITS');
      expect(JSON.parse(result[1].text).type).toBe('CLOSING_CREDITS');
```